File: samples/dotnet-csharp-graphdocs/ConnectionService.cs
static class ConnectionService
{
  async static Task CreateConnection()
  {
    Console.Write("Creating connection...");

    await GraphService.Client.External.Connections
      .PostAsync(ConnectionConfiguration.ExternalConnection);

    Console.WriteLine("DONE");
  }

  async static Task CreateSchema()
  {
    Console.WriteLine("Creating schema...");

    await GraphService.Client.External
      .Connections[ConnectionConfiguration.ExternalConnection.Id]
      .Schema
      .PatchAsync(ConnectionConfiguration.Schema);

    Console.WriteLine("DONE");
  }

  public static async Task ProvisionConnection()
  {
    try
    {
      await CreateConnection();
      await CreateSchema();
    }
    catch (Exception ex)
    {
      Console.WriteLine(ex.Message);
    }
  }
}

File: samples/java-markdown/app/src/main/java/connector/ConnectionService.java
package connector;

import java.io.IOException;
import java.util.concurrent.ExecutionException;

public class ConnectionService {
    private static void CreateConnection() throws InterruptedException, ExecutionException, IOException {
        System.out.print("Creating connection...");

        GraphService.getClient().external().connections()
                .buildRequest()
                .postAsync(ConnectionConfiguration.getExternalConnection()).get();

        System.out.println("DONE");
    }

    private static void CreateSchema() throws InterruptedException, ExecutionException, IOException {
        System.out.println("Creating schema...");

        GraphService.getClient().external()
                .connections(ConnectionConfiguration.getExternalConnection().id)
                .schema()
                .buildRequest()
                .patchAsync(ConnectionConfiguration.getSchema()).get();

        System.out.println("DONE");
    }

    public static void provisionConnection() {
        try {
            CreateConnection();
            CreateSchema();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}
